File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
import java.io.*;
class Solution {
    public int solution(int[][] sizes) {
        int width = 0;
        int height = 0;
        for(int[] card: sizes){
            int w = 0;
            int h = 0;
            w = Math.max(card[0], card[1]);
            h = Math.min(card[0], card[1]);
            width = Math.max(width, w);
            height = Math.max(height, h);
        }
        return width * height;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0 # 모든 명함 수납할 수 있는 최소 크기
    min_size, max_size, w, h = 1001, 0, 0, 0
    for size in sizes:
        min_size = (min(size[0], size[1]))
        max_size = (max(size[1], size[0]))
        w = max(w, min_size)
        h = max(h, max_size)
        
    return w*h